src/main/java/nl/tudelft/contextproject/model/level/MazeTile.java

```diff
@@ -30,63 +35,85 @@
 	 * 			the y coordinate of this tile
 	 * @param type
 	 * 			the type of this tile
-     */
+	 */
 	public MazeTile(int x, int y, TileType type) {
 		this.position = new Vector2f(x, y);
 		this.explored = false;
-
+		this.type = type;
+		
 		switch (type) {
-			case FLOOR:
-				this.height = 0;
-				this.color = ColorRGBA.LightGray;
-				break;
-			case WALL:
-				this.height = 3;
-				this.color = ColorRGBA.Blue;
-				break;
-			case CORRIDOR:
-				this.height = 0;
-				this.color = ColorRGBA.DarkGray;
-				break;
-			default:
-				throw new IllegalArgumentException("Invalid TileType: " + type);
+		case FLOOR:
+			this.height = 0;
+			this.color = ColorRGBA.LightGray;
+			break;
+		case WALL:
+			this.height = 3;
+			this.color = ColorRGBA.Blue;
+			break;
+		case CORRIDOR:
+			this.height = 0;
+			this.color = ColorRGBA.DarkGray;
+			break;
+		default:
+			throw new IllegalArgumentException("Invalid TileType: " + type);
 		}
 	}
 	
+	/**
+	 * Getter for tileType.
+	 * @return
+	 * 				type of maze tile
+	 */
+	public TileType getTileType() {
+		return type;
+	}
+
 	@Override
 	public Spatial getSpatial() {
-        if (spatial != null) return spatial;
-        
-        Box b = new Box(.5f, .5f, .5f + height); // create cube shape
-        this.spatial = new Geometry("Box", b);  // create cube geometry from the shape
-        Material mat = new Material(Main.getInstance().getAssetManager(), "Common/MatDefs/Light/Lighting.j3md");  // create a simple material
-        mat.setBoolean("UseMaterialColors", true);    
-        mat.setColor("Diffuse", color);
-        mat.setColor("Specular", ColorRGBA.White);
-        mat.setFloat("Shininess", 64f);  // [0,128]
+		if (spatial != null) return spatial;
+
+		Box b = new Box(.5f, .5f + height, .5f); // create cube shape
+		this.spatial = new Geometry("Box", b);  // create cube geometry from the shape
+		Material mat = new Material(Main.getInstance().getAssetManager(), "Common/MatDefs/Light/Lighting.j3md");  // create a simple material
+		mat.setBoolean("UseMaterialColors", true);    
+		mat.setColor("Diffuse", color);
+		mat.setColor("Specular", ColorRGBA.White);
+		mat.setFloat("Shininess", 64f);  // [0,128]
 		mat.setColor("Ambient", color);
-        this.spatial.setMaterial(mat);                   // set the cube's material
-        this.spatial.move(position.x, position.y, height);
-        return spatial;
+		this.spatial.setMaterial(mat);                   // set the cube's material
+		this.spatial.move(position.x, height, position.y);
```

Comment:
Well caught, this small parameter mess up could have caused a ton more problems...